pubsub-protocol-binding.md

```diff
@@ -0,0 +1,276 @@
+to Pub/Sub attributes.
+
+### 1.4. Event Formats
+
+Event formats, used with the _structured_ content mode, define how an event is
+expressed in a particular data format. All implementations of this
+specification MUST support the [JSON event format][json-format], but MAY support
+any additional, including proprietary, formats.
+
+### 1.5. Security
+
+This specification does not introduce any new security features for Pub/Sub, or
+mandate specific existing features to be used.
+
+## 2. Use of CloudEvents Attributes
+
+This specification does not further define any of the [CloudEvents][ce] event
+attributes.
+
+This mapping is intentionally robust against changes, including the addition and
+removal of event attributes, and also accommodates vendor extensions to the
+event metadata.
+
+### 2.1. datacontenttype Attribute
+
+The `datacontenttype` attribute is assumed to contain a [RFC2046][rfc2046]
+compliant media-type expression.
+
+### 2.2. data Attribute
+
+The `data` attribute is assumed to contain opaque application data that is
+encoded as declared by the `datacontenttype` attribute.
+
+An application is free to hold the information in any in-memory representation
+of its choosing, but as the value is transposed into a Pub/Sub message as defined 
+in this specification, the assumption is that the `data` attribute value is made
+available as a sequence of bytes.
+
+For instance, if the declared `datacontenttype` is
+`application/json;charset=utf-8`, the expectation is that the `data` attribute
+value is made available as [UTF-8][rfc3629] encoded JSON text Pub/Sub message data.
+
+## 3. Pub/Sub Message Mapping
+
+With Pub/Sub, the content mode is chosen by the sender of the
+event. Protocol usage patterns that might allow solicitation of events using a
+particular content mode might be defined by an application, but are not defined
+here.
+
+The receiver of the event can distinguish between the two content modes by
+inspecting the [Pub/Sub Attribute `Content-Type`][content-type] of the
+[Pub/Sub message][pubsub-message]. If the attribute is present and its value is prefixed 
+with the CloudEvents media type `application/cloudevents`, indicating the use of a known
+[event format](#14-event-formats), the receiver uses _structured_ mode,
+otherwise it defaults to _binary_ mode.
+
+If a receiver finds a CloudEvents media type as per the above rule, but with an
+event format that it cannot handle, for instance
+`application/cloudevents+avro`, it MAY still treat the event as binary and
+forward it to another party as-is.
+
+### 3.1. Binary Content Mode
+
+The _binary_ content mode accommodates any shape of event data, and allows for
+efficient transfer and without transcoding effort.
+
+#### 3.1.1. Content-Type
+
+For the _binary_ mode, the [Pub/Sub Attribute `Content-Type`][content-type] of the 
+[Pub/Sub message][pubsub-message] property, if present, MUST be mapped
+directly to the CloudEvents `datacontenttype` attribute.
```

Comment:
> "The Pub/Sub Attribute `Content-Type` of the Pub/Sub message property

What is the message property? That term isn't defined or used anywhere else. I think this can just be "The Pub/Sub Attribute `Content-Type`", dropping the last 5 words referring to the property.

> MUST be mapped directly to

What does it mean to be "mapped directly to"? Must the values be identical? Must they have a deterministic transformation between them? This should be clearer.

Reply:
removed...

I ended up putting MUST be used... we shouldn't be transforming it...